lib/rules/no-unregistered-components.js

```diff
@@ -0,0 +1,134 @@
+
+const utils = require('eslint-plugin-vue/lib/utils')
+const casing = require('eslint-plugin-vue/lib/utils/casing')
+
+// ------------------------------------------------------------------------------
+// Rule helpers
+// ------------------------------------------------------------------------------
+
+const VUE_BUILT_IN_COMPONENTS = [
+  'component',
+  'suspense',
+  'teleport',
+  'transition',
+  'transition-group',
+  'keep-alive',
+  'slot'
+]
+/**
+ * Check whether the given node is a built-in component or not.
+ *
+ * Includes `suspense` and `teleport` from Vue 3.
+ *
+ * @param {ASTNode} node The start tag node to check.
+ * @returns {boolean} `true` if the node is a built-in component.
+ */
+const isBuiltInComponent = (node) => {
+  const rawName = node && casing.kebabCase(node.rawName)
+  return utils.isHtmlElementNode(node) &&
+    !utils.isHtmlWellKnownElementName(node.rawName) &&
+    VUE_BUILT_IN_COMPONENTS.indexOf(rawName) > -1
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow using components that are not registered inside templates',
+      categories: null,
+      recommended: false,
+      url: 'https://eslint.vuejs.org/rules/no-unregistered-components.html'
+    },
+    fixable: null,
+    schema: [{
+      type: 'object',
+      properties: {
+        ignorePatterns: {
+          type: 'array'
+        }
+      },
+      additionalProperties: false
+    }]
+  },
+
+  create (context) {
+    const options = context.options[0] || {}
+    const ignorePatterns = options.ignorePatterns || []
+    const usedComponentNodes = []
+    const registeredComponents = []
+    let templateLocation
+
+    return utils.defineTemplateBodyVisitor(context, {
+      VElement (node) {
+        if (
+          (!utils.isHtmlElementNode(node) && !utils.isSvgElementNode(node)) ||
+          utils.isHtmlWellKnownElementName(node.rawName) ||
+          utils.isSvgWellKnownElementName(node.rawName) ||
+          isBuiltInComponent(node)
+        ) {
+          return
+        }
+
+        usedComponentNodes.push({ node, name: node.rawName })
+      },
+      "VAttribute[directive=true][key.name.name='bind'][key.argument.name='is']" (node) {
+        if (
+          !node.value ||
+          node.value.type !== 'VExpressionContainer' ||
+          !node.value.expression
+        ) return
+
+        if (node.value.expression.type === 'Literal') {
+          if (utils.isHtmlWellKnownElementName(node.value.expression.value)) return
+          usedComponentNodes.push({ node, name: node.value.expression.value })
+        }
+      },
+      "VAttribute[directive=false][key.name='is']" (node) {
+        if (utils.isHtmlWellKnownElementName(node.value.value)) return
```

Comment:
`node.value` may be null. (I just noticed this.)

e.g.

```vue
<component is />
```

Could you add this test case and change the source?

https://github.com/mysticatea/vue-eslint-parser/blob/master/docs/ast.md#vattribute